frontend/packages/console-app/src/components/cloud-shell/AutoResizeTerminal.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import { Terminal as XTerminal } from 'xterm';
+import Measure from 'react-measure';
+import * as fit from 'xterm/lib/addons/fit/fit';
+import * as full from 'xterm/lib/addons/fullscreen/fullscreen';
+
+XTerminal.applyAddon(fit);
+XTerminal.applyAddon(full);
+
+type TerminalProps = {
+  onData: any;
+  className?: string;
+};
+
+type TerminalState = {
+  width: number;
+  height: number;
+};
+
+export class Terminal extends React.Component<TerminalProps, TerminalState> {
+  private innerRef;
+
+  private outerRef;
+
+  private onDataReceived;
+
+  private terminal;
+
+  private padding;
+
+  constructor(props) {
+    super(props);
+    this.innerRef = React.createRef();
+    this.outerRef = React.createRef();
+    this.state = {
+      width: 0,
+      height: 0,
+    };
+    this.onDataReceived = (data) => {
+      this.terminal && this.terminal.write(data);
+    };
+    this.padding = 20;
+    const options = {
+      fontFamily: 'monospace',
+      fontSize: 16,
+      cursorBlink: false,
+      cols: 80,
+      rows: 25,
+    };
+    this.terminal = new XTerminal(Object.assign({}, options));
+    this.terminal.on('data', this.props.onData);
+  }
+
+  focus = () => {
+    this.terminal && this.terminal.focus();
+  };
+
+  onResize = () => {
```

Comment:
Fairly certain nothing in this function is required.

Reply:
altered to a fit() call in a try catch scenario